scripts/experiments_tabular/maf.jl

```diff
@@ -0,0 +1,92 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+using StatsBase: fit!, predict, sample
+using BSON
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+        required = true
+        arg_type = Int
+        help = "seed"
+    "dataset"
+        required = true
+        arg_type = String
+        help = "dataset"
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed = parsed_args
+
+modelname = "MAF"
+function sample_params()
+	par_vec = ([2, 5, 10], 2 .^(4:10), 2:3, ["natural", "random"], [1f-4], [100], [30], [1f-6])
+	argnames = (:nflows, :hdim, :nlayers, :ordering, :lr, :batchsize, :patience, :wreg)
+
+	return (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
+end
+
+function fit(data, parameters)
+	idim = size(data[1][1], 1)
+
+	model = GenerativeAD.Models.MAF(
+				parameters.nflows,
+				idim,
+				parameters.hdim,
+				parameters.nlayers,
+				parameters.ordering)
+
+	try
+		global info, fit_t, _, _, _ = @timed fit!(model, data, parameters)
+ 	catch e
+		@info "Failed training due to \n$e"
+		return (fit_t = NaN,), []
+	end
+
+	training_info = (
+		fit_t = fit_t,
+		history = info.history,
+		niter = info.niter,
+		npars = info.npars,
+		model = info.model
+		)
+
+	training_info, [(x -> predict(info.model, x), parameters)]
+end
+
+
+try_counter = 0
+max_tries = 10*max_seed
+while try_counter < max_tries
+    parameters = sample_params()
+
+    for seed in 1:max_seed
+		savepath = datadir("experiments/tabular/$(modelname)/$(dataset)/seed=$(seed)")
+		mkpath(savepath)
+
+		data = GenerativeAD.load_data(dataset, seed=seed)
+		edited_parameters = GenerativeAD.edit_params(data, parameters)
+
+		if GenerativeAD.check_params(savepath, data, edited_parameters)
+			@info "Started training $(modelname)$(edited_parameters) on $(dataset):$(seed)"
+			@info "Train/valdiation/test splits: $(size(data[1][1], 2)) | $(size(data[2][1], 2)) | $(size(data[2][1], 2))"
+			@info "Number of features: $(size(data[1][1], 1))"
+			
+			training_info, results = fit(data, edited_parameters)
+
+			tagsave(joinpath(savepath, savename("model", parameters, "bson")), Dict("model"=>training_info.model), safe = true)
```

Comment:
This is gonna fail in case of error in fit, since there is no field model. Consider checking `if training_info.model != nothing`

Reply:
Good catch.